Lib/argparse.py

```diff
@@ -1697,6 +1697,20 @@ def add_mutually_exclusive_group(self, *args, **kwargs):
         return super().add_mutually_exclusive_group(*args, **kwargs)
 
 
+def _prog_name(prog=None):
+    if prog is not None:
+        return prog
+    arg0 = _sys.argv[0]
+    modspec = _sys.modules['__main__'].__spec__
+    if modspec is None:
+        return _os.path.basename(arg0)
+    py = _os.path.basename(_sys.executable)
+    if (modspec.name == '__main__' and not modspec.parent and modspec.has_location
+        and _os.path.dirname(modspec.origin) == _os.path.join(_os.getcwd(), arg0)):
```

Comment:
Actually, it seems that this condition can be simplified to simply `modspec.name == '__main__'`). What are your thoughts @ncoghlan?

Reply:
Actually, additionally checks were just an obstacle, because of differences in normalization between Unix and Windows. So I removed them.